src/Microsoft.Framework.Configuration/ConfigurationBase.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.Framework.Configuration
+{
+    public abstract class ConfigurationBase : IConfiguration
```

Comment:
This seems like a weird name, why isn't this ConfigurationElement or Node?  Base just doesn't seem to add any meaning